sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -301,9 +331,83 @@ class CAFMaker : public art::EDProducer {
   // setup volume definitions
   InitVolumes();
 
-  // setup random number generator
+  // setup random number generators
   fFakeRecoTRandom = new TRandomMT64(art::ServiceHandle<rndm::NuRandomService>()->getSeed());
+  fBlindTRandom = new TRandomMT64(art::ServiceHandle<rndm::NuRandomService>()->getSeed());
+}
+
+//......................................................................
+  double CAFMaker::GetBlindPOTScale() const {
+   std::string bstring = std::to_string(fParams.POTBlindSeed());
+   int slen = bstring.length();
+   std::string s1 = bstring.substr(0,int(slen/2));
+   std::string s2 = bstring.substr(int(slen/2));
+   double rat = stod(s1)/stod(s2);
+   while (abs(rat)>1){
+     rat = -1 * (abs(rat) - 1);
+   }
+   return 1 + rat*0.3;
+   
+  }
+//......................................................................
+void CAFMaker::BlindEnergyParameters(StandardRecord* brec) {
+
+  //simple cuts for trk and shower variables
+  //blind events with a potential lepton with momentum > 0.6 that starts in fiducial volume
+  for (unsigned int i=0; i<brec->reco.ntrk; ++i) {
+    if ( ((brec->reco.trk[i].start.x < -71.1 - 25 && brec->reco.trk[i].start.x > -369.33 + 25 ) ||
+	  (brec->reco.trk[i].start.x > 71.1 + 25 && brec->reco.trk[i].start.x < 369.33 - 25 )) &&
+	 (brec->reco.trk[i].start.y > -181.7 + 25 && brec->reco.trk[i].start.y < 134.8 - 25 ) &&
+	 (brec->reco.trk[i].start.z  > -895.95 + 30 && brec->reco.trk[i].start.z < 895.95 - 50)) {
+
+      if (brec->reco.trk[i].mcsP.fwdP_muon > 0.6) {
+	brec->reco.trk[i].mcsP.fwdP_muon = TMath::QuietNaN();    
+      }
+      if (brec->reco.trk[i].rangeP.p_muon > 0.6) {
+	brec->reco.trk[i].rangeP.p_muon = TMath::QuietNaN();
+      }
+    }
+  }
```

Comment:
```suggestion
  for (const caf::SRTrack& trk: brk->reco.trk) {
    const caf::SRVector3D start = trk.start;
    const caf::SRVector3D end = trk.end;
    if ( ((start.x < -71.1 - 25 && start.x > -369.33 + 25 ) ||
	  (start.x > 71.1 + 25 && start.x < 369.33 - 25 )) &&
	 (start.y > -181.7 + 25 && start.y < 134.8 - 25 ) &&
	 (start.z  > -895.95 + 30 && start.z < 895.95 - 50)) {

      if (trk.mcsP.fwdP_muon > 0.6) {
	trk.mcsP.fwdP_muon = TMath::QuietNaN();    
      }
      if (trk.rangeP.p_muon > 0.6) {
	trk.rangeP.p_muon = TMath::QuietNaN();
      }
    }
  }
```
(untested)

Reply:
Had to remove const from "const caf::SRTrack& trk: brk->reco.trk" for this to compile since I'm later using trk to change the momemtum to nan.